yesod-core/Yesod/Core/Class/Yesod.hs

```diff
@@ -627,81 +627,102 @@ widgetToPageContent w = do
 
 -- | The default error handler for 'errorHandler'.
 defaultErrorHandler :: Yesod site => ErrorResponse -> HandlerT site IO TypedContent
-defaultErrorHandler NotFound = selectRep $ do
-    provideRep $ defaultLayout $ do
-        r <- waiRequest
-        let path' = TE.decodeUtf8With TEE.lenientDecode $ W.rawPathInfo r
-        setTitle "Not Found"
-        toWidget [hamlet|
-            <h1>Not Found
-            <p>#{path'}
-        |]
-    provideRep $ return $ object ["message" .= ("Not Found" :: Text)]
-
--- For API requests.
--- For a user with a browser,
--- if you specify an authRoute the user will be redirected there and
--- this page will not be shown.
-defaultErrorHandler NotAuthenticated = selectRep $ do
-    provideRep $ defaultLayout $ do
-        setTitle "Not logged in"
-        toWidget [hamlet|
-            <h1>Not logged in
-            <p style="display:none;">Set the authRoute and the user will be redirected there.
-        |]
+defaultErrorHandler resp = case resp of
```

Comment:
Would it be possible to switch back to the original function pattern matching in place of the case statement, and just move the `where` clause definitions to top level functions? It would make it much easier to see in the history what the actual change here was.